project-base/storefront/components/Blocks/Banners/BannersSlider.tsx

```diff
@@ -1,102 +1,120 @@
+import { BannerImage } from './BannerImage';
+import { bannersReducer, getBannerOrderCSSProperty } from './bannersUtils';
 import { TriangleIcon } from 'components/Basic/Icon/TriangleIcon';
-import { BannersSliderItem } from 'components/Blocks/Banners/BannersSliderItem';
 import { TIDs } from 'cypress/tids';
 import { TypeSliderItemFragment } from 'graphql/requests/sliderItems/fragments/SliderItemFragment.generated';
-import 'keen-slider/keen-slider.min.css';
-import { useKeenSlider } from 'keen-slider/react';
-import { useEffect, useRef, useState } from 'react';
-import Skeleton from 'react-loading-skeleton';
-import { desktopFirstSizes } from 'utils/mediaQueries';
-import { useGetWindowSize } from 'utils/ui/useGetWindowSize';
+import { useEffect, useReducer, useRef } from 'react';
+import { useSwipeable } from 'react-swipeable';
+import { twJoin } from 'tailwind-merge';
 
 export type BannersSliderProps = {
     sliderItems: TypeSliderItemFragment[];
 };
 
 export const BannersSlider: FC<BannersSliderProps> = ({ sliderItems }) => {
-    const [currentSlide, setCurrentSlide] = useState(0);
-    const [pause, setPause] = useState(false);
-    const timer = useRef<NodeJS.Timeout | null>(null);
-    const sliderBoxRef = useRef<HTMLDivElement>(null);
+    const numItems = sliderItems.length;
+    const [bannerSliderState, dispatchBannerSliderStateChange] = useReducer(bannersReducer, {
+        sliderPosition: 0,
+        isSliding: false,
+        slideDirection: 'NEXT',
+    });
+    const intervalRef = useRef<NodeJS.Timeout | null>(null);
 
-    const { width: windowWidth } = useGetWindowSize();
-    const isDesktop = windowWidth > desktopFirstSizes.tablet;
-    const isRecognizingWindowWidth = windowWidth < 0;
+    const slide = (dir: 'PREV' | 'NEXT') => {
+        dispatchBannerSliderStateChange({ type: dir, numItems });
+        setTimeout(() => {
+            dispatchBannerSliderStateChange({ type: 'STOP_SLIDING' });
+        }, 50);
+    };
 
-    const [sliderRef, slider] = useKeenSlider<HTMLDivElement>({
-        loop: true,
-        duration: 1000,
-        slidesPerView: 1,
-        autoAdjustSlidesPerView: isRecognizingWindowWidth,
-        slideChanged: (slider) => {
-            setCurrentSlide(slider.details().relativeSlide);
-        },
-        dragStart: () => {
-            setPause(true);
-        },
-        dragEnd: () => {
-            setPause(false);
-        },
+    const handlers = useSwipeable({
+        onSwipedLeft: () => slide('NEXT'),
+        onSwipedRight: () => slide('PREV'),
+        swipeDuration: 500,
```

Comment:
Same as the slider interval comment below. For simpler readability and customization.